skglm/solvers/prox_newton.py

```diff
@@ -86,7 +92,11 @@ def solve(self, X, y, datafit, penalty, w_init=None, Xw_init=None):
             else:
                 grad = _construct_grad(X, y, w[:n_features], Xw, datafit, all_features)
 
-            opt = penalty.subdiff_distance(w[:n_features], grad, all_features)
+            if self.ws_strategy == "subdiff":
```

Comment:
``fixpoint`` should be also considered when determining the descent direction

https://github.com/scikit-learn-contrib/skglm/blob/4cf5976b11563e49e76c4b880521bdd00e2e8fa8/skglm/solvers/prox_newton.py#L218